### ISSUE TITLE
[no-unreachable] and [no-unused-label] are handled inconsistently

### ISSUE THREAD

**Issue by amayer42** [typescript-eslint/typescript-eslint#4819]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
function test() {
  return 'test';
  const a = 'a';
}

Label: {
  test();
}
```

**Expected Result**

The unreachable code and the unused label should both be handled by either TypeScript or by the `eslint:recommended` rules.

**Actual Result**

With default settings, the unused label is flagged as invalid both by VSCode (dims the text and provides the reason: "Unused label.ts(7028)") and when linting ("'Label:' is defined but never used.eslint([no-unused-labels](https://eslint.org/docs/rules/no-unused-labels))"). The unreachable code is only flagged as invalid by VSCode (dims the text and provides the reason: "Unreachable code detected.ts(7027)").

**Additional Info**

Based upon my understanding of the eslint-recommended, any rule that is handled by the TypeScript compiler by default should be disabled by this config. I'm curious if this stance holds even for the rules above that are not considered errors in TypeScript by default. 

- https://www.typescriptlang.org/tsconfig/#allowUnreachableCode
- https://www.typescriptlang.org/tsconfig/#allowUnusedLabels

It might be good to document that stance a bit better. That seems like the decision from #1041, though it seems that nothing was ever done to enhance the documentation.

Documentation aside, I think that it at least makes sense to be as consistent as possible with how the eslint-recommended config handles rules that overlap with TypeScript.

**Versions**

It's worth pointing out along with these versions that as of `v5.19.0`, the rules in question [still have the same settings as in my local versions](https://github.com/typescript-eslint/typescript-eslint/blob/v5.19.0/packages/eslint-plugin/src/configs/eslint-recommended.ts)

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.31.0` |
| `node`                             | `14.19.0` |

**Comment by JoshuaKGoldberg** [comment]
Once #4706 is added with overhauled preset config docs, it'd be good to document this a bit more. +1

**Comment by tylerlaprade** [comment]
I'm also having this problem. I enabled `allowUnreachableCode` in TypeScript because it was automatically "fixing" my code by deleting it when I was in the middle of a refactor. I had to manually add `no-unreachable` to my ESLint rules because even though it's set in `eslint-recommended`, it is disabled in `"plugin:@typescript-eslint/strict".

**Comment by bradzacher** [comment]
> even though it's set in eslint-recommended, it is disabled in `"plugin:@typescript-eslint/strict".

It's turned off in `eslint-recommended`, and `strict` extends that config - so by extension it also disables it.

> because it was automatically "fixing" my code by deleting it when I was in the middle of a refactor

TS doesn't autofix delete code ever and the `no-unreachable` ESLint rule doesn't either. If your code was being deleted then it was some other tool in your chain.

**Comment by tylerlaprade** [comment]
Hmm, I'm not having that problem anymore. Possibly there was a bug in one of the tools at that time.

**Comment by JoshuaKGoldberg** [comment]
It's been >2 years, only one person has 👍d the issue,  and nobody has sent a PR. I'm going to close it out now as something the community hasn't given us a lot of noise on wanting. If anybody thinks I'm wrong to do this, please yell at me. 🙂
